File: data/java/src/function_Schematics_model_to_native.java
package com.example;

import com.google.common.base.Preconditions;

public class Person {
    public String get_native_representation(String data) {
        // 检查输入格式是否正确（不能为空且必须包含两个逗号分隔的部分）
        if (data == null || data.isEmpty() || data.split(",").length != 2) {
            return "error: Invalid input format";
        }

        String[] parts = data.split(",", -1);
        String name = parts[0];
        int age;
        try {
            age = Integer.parseInt(parts[1]);
        } catch (NumberFormatException e) {
            return "error: Invalid input format";
        }

        // 使用Guava Preconditions进行数据验证
        try {
            Preconditions.checkNotNull(name, "name is required");
            Preconditions.checkArgument(age >= 0, "age must be greater than or equal to 0");
        } catch (IllegalArgumentException e) {
            return "error: " + e.getMessage();
        }

        // 默认城市为 "Unknown"
        String city = "Unknown";

        return name + " " + age + " " + city;
    }
}

File: manual/data/python/src/function_Schematics_model_to_native.py
from schematics.models import Model
from schematics.types import StringType, IntType
from schematics.exceptions import DataError

class Person:
    def get_native_representation(self, data: str) -> str:
        if not data or len(data.split(',')) != 2:
            return "error: Invalid input format"
        
        data = data.split(",")
        data_dict = {"name": data[0], "age": int(data[1])}
        
        class PersonModel(Model):
            name = StringType(required=True)
            age = IntType(required=True, min_value=0)
            city = StringType(default="Unknown")
        
        person = PersonModel(data_dict)
        try:
            person.validate()
            return f"{person.name} {person.age} {person.city}"
        except DataError as e:
            return f"error: {e.messages}"
